rllib/core/rl_trainer/rl_trainer.py

```diff
@@ -140,9 +139,11 @@ def __init__(
         self.module_spec = module_spec
         self.module_obj = module
         self.optimizer_config = optimizer_config
-        self.distributed = distributed
-        self.scaling_config = scaling_config
-        self.config = algorithm_config
+        self.config = trainer_hyperparameters
+
+        # pick the configs that we need for the trainer from scaling config
+        scaling_config = scaling_config or RLTrainerScalingConfig()
```

Comment:
The base class now accepts this parameter for any possible model distribution config, and the constructor will decide which values from that config to store. (similar to construction of connectors)